ui/components/MesheryPerformance/Dashboard.js

```diff
@@ -82,6 +120,12 @@ function Dashboard({ updateProgress, enqueueSnackbar, closeSnackbar, grafana, ro
     fetchTests();
   }, []);
 
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      setCounter(counter - 1)
+    }, 1000)
+    return () => clearTimeout(timer)
+  })
```

Comment:
The useEffect Body runs on every state change which is not desired. Actually this modal can be placed in it's own functional component for the purpose of re-usability and then this implementation will look good.

Reply:
done